packages/cicero-core/src/instance.js

```diff
@@ -25,60 +26,82 @@ const HtmlTransformer = require('@accordproject/markdown-html').HtmlTransformer;
 // For formulas evaluation
 const ErgoEngine = require('@accordproject/ergo-engine/index.browser.js').EvalEngine;
 
+const Util = require('./util');
+
 /**
- * A TemplateInstance is an instance of a Clause or Contract template. It is executable business logic, linked to
+ * A Instance is an instance of a Clause or Contract template. It is executable business logic, linked to
  * a natural language (legally enforceable) template.
- * A TemplateInstance must be constructed with a template and then prior to execution the data for the clause must be set.
- * Set the data for the TemplateInstance by either calling the setData method or by
+ * A Instance must be constructed with a template and then prior to execution the data for the clause must be set.
+ * Set the data for the Instance by either calling the setData method or by
  * calling the parse method and passing in natural language text that conforms to the template grammar.
  * @public
  * @abstract
  * @class
  */
-class TemplateInstance {
-
+class Instance {
     /**
-     * Create the Clause and link it to a Template.
-     * @param {Template} template  - the template for the clause
+     * Create an instance
+     * @param {object} metadata - the metadata object
+     * @param {number} logicManager - the logic manager
+     * @param {string} grammar - the initial grammar
+     * @param {Template} [template] - the template for the instance
+     * @param {string} instantiator - name of the person/party which instantiates the contract instance
      */
-    constructor(template) {
-        if (this.constructor === TemplateInstance) {
-            throw new TypeError('Abstract class "TemplateInstance" cannot be instantiated directly.');
+    constructor(metadata, logicManager, grammar, template, instantiator) {
+        if (this.constructor === Instance) {
+            throw new TypeError('Abstract class "Instance" cannot be instantiated directly.');
         }
+
+        this.metadata = metadata;
+        this.instanceKind = metadata.getTemplateType();
+        this.prefix = metadata.getIdentifier();
+        this.logicManager = logicManager;
+        this.runtime = metadata.getRuntime();
         this.template = template;
-        this.data = null;
-        this.concertoData = null;
+        this.instantiator = instantiator;
+
         this.ciceroMarkTransformer = new CiceroMarkTransformer();
         this.templateMarkTransformer = new TemplateMarkTransformer();
-        this.parserManager = this.template.getParserManager();
+        this.parserManager = new ParserManager(this.logicManager.getModelManager(), null, this.instanceKind);
```

Comment:
Why aren't we getting the parser manager from the template anymore?

Reply:
@dselman - @sanketshevkar thinks that this might be a question for @jeromesimeon (??)